### ISSUE TITLE
[Enhancement] Open With Program LIsts

### ISSUE THREAD

**Issue by nonentity1001** [Sigil-Ebook/Sigil#390]
Maintain a list of previously used editors for each file type (xhtml, jpeg, mp4, etc), rather than just the most recently used editor for each file type. Sometimes I can't recall which tool I used for some obscure task, just seeing a list of names would jog the neurons - particularly with non text editors.

See https://www.mobileread.com/forums/showthread.php?t=314422

Also see https://www.mobileread.com/forums/showpost.php?p=3799891&postcount=5

FYI - I'm not subscribed to this thread

**Comment by kevinhendricks** [comment]
Okay, I just pushed very very preliminary support to master for keeping up to 5 editors for each resource type.

If this turns out to be stable and not cause problems.  It should be in the last release.
The list is always ordered by the most recently used app for that resource type first.

I used new settings keys so that none of the older open_with settings are touched so if someone needs to drop back a Sigil version, all should still be good.
